### PR TITLE
[RectCore] fix auth logout by invalidating and storing invalidated cookie

### LINKED ISSUE #762: CORS issue for /auth/logout
Every other auth hook uses `credentials: "include"` (to account for CORS) in requests to the backend, except for logout hook.
This leads to a situation, where the server's attempt to clear cookie on logout is completely ignored by the browser.

`./packages/react-core/src/evm/hooks/auth/useUser.ts` should use the same approach to requests, just like every other hook in that folder.

### PR DESCRIPTION
**Pull request by lucoadam**
- Fixes #762

### CODE REVIEW

File: .changeset/beige-plums-hug.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@thirdweb-dev/react-core": minor
```
Comment (jnsdls) [inline .changeset/beige-plums-hug.md:2]:
```suggestion
"@thirdweb-dev/react-core": patch
```

**Review by jnsdls — COMMENTED** [review]
thanks for the PR @lucoadam - I think this should be a `patch` change since it's a bug fix

**Comment by jnsdls** [comment]
@adam-maj can u take a look?

**Comment by lucoadam** [comment]
> thx again @lucoadam!

Thank you guys for the awesome product

**Pull request merged**